subsys/bluetooth/controller/ll_sw/nordic/lll/lll_adv_iso.c

```diff
@@ -242,27 +242,26 @@ static int prepare_cb_common(struct lll_prepare_param *p)
 	lll_chan_set(data_chan_use);
 
 	/* FIXME: get ISO data PDU */
-	if (0) {
-		pdu = (void *)radio_pkt_empty_get();
-		pdu->ll_id = PDU_BIS_LLID_START_CONTINUE;
-		pdu->len = 0U;
-	} else {
-		pdu = radio_pkt_scratch_get();
+	pdu = radio_pkt_scratch_get();
+	if (lll->bn_curr >= lll->bn) {
 		pdu->ll_id = PDU_BIS_LLID_COMPLETE_END;
-		pdu->len = LL_BIS_OCTETS_TX_MAX;
+	} else {
+		pdu->ll_id = PDU_BIS_LLID_START_CONTINUE;
+	}
+	pdu->len = LL_BIS_OCTETS_TX_MAX;
 
-		pdu->payload[0] = lll->bn_curr;
-		pdu->payload[1] = lll->irc_curr;
-		pdu->payload[2] = lll->ptc_curr;
-		pdu->payload[3] = lll->bis_curr;
+	pdu->payload[0] = lll->bn_curr;
+	pdu->payload[1] = lll->irc_curr;
+	pdu->payload[2] = lll->ptc_curr;
+	pdu->payload[3] = lll->bis_curr;
```

Comment:
These are almost identical to lines 454-467.

1) Can they be moved to a common function?
2) In the lines below, we also we set
```
		pdu->cssn = lll->cssn;
		pdu->cstf = 0U;
```
Should that be done here as well?

Reply:
1. The PDU populated here is test data which otherwise is to be prepared by ISOAL Tx. It will be removed in the future.
2. `cstf` and `cssn` is filled here: https://github.com/zephyrproject-rtos/zephyr/pull/40502/files#diff-d8ee5f132634e6624acd0487f6cb124b3422ece71a5843082fb9b38a454ee465R268-R288